File: backend-dotnet/Application/Queries/Posts/GetPublicPostsQuery.cs
using Application.DTOs;

namespace Application.Queries.Posts
{
    public class GetPublicPostsQuery : IQuery<PagedResult<PostListItemDto>>
    {
        public Guid? CurrentUserId { get; set; }
        public int Page { get; set; } = 1;
        public int PageSize { get; set; } = 10;
    }
}

File: backend-java/src/main/java/com/jerrygram/application/queries/posts/GetPublicPostsQuery.java
package com.jerrygram.application.queries.posts;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class GetPublicPostsQuery {
    private UUID currentUserId;
    private int page;
    private int pageSize;
}